R/git.R

```diff
@@ -125,60 +125,121 @@ git_blame <- function(git_repo_path,flags,commit_hash,file_path){
+  #folder_path <- file.path(folder_path,"kaiaulu_sample")
+  folder_path <- io_create_folder("kaiaulu_sample")
 
   #mkdir path/to/folder/sample
-  error <- system2('mkdir',
-                   args = c(folder_path),
-                   stdout = TRUE,
-                   stderr = FALSE)
+  #error <- system2('mkdir',
+  #                 args = c(folder_path),
+  #                 stdout = TRUE,
+  #                 stderr = FALSE)
 
 
   file_path <- file.path(folder_path,"hello.R")
 
+  io_make_sample_file(file_path, "print('hello world!')")
+
   #echo "print('hello world!')" >  path/to/folder/hello.R
-  error <- system2('echo',
-                   args = c("\"print('hello world!')\"",
-                            '>',
-                            file_path),
-                   stdout = TRUE,
-                   stderr = FALSE)
+  #error <- system2('echo',
+  #                 args = c("\"print('hello world!')\"",
+  #                          '>',
+  #                          file_path),
+  #                 stdout = TRUE,
+  #                 stderr = FALSE)
 
-  # git init path/to/folder
-  error <- system2('git',
-                   args = c('init',
-                            folder_path),
-                   stdout = TRUE,
-                   stderr = FALSE)
+  git_init(folder_path)
 
   git_repo <- file.path(folder_path,'.git')
 
-  # git --git-dir sample/.git --work-tree sample add hello.R
+  git_add(git_repo, folder_path, file_path)
+
+  # WILL UNCOMMENT THIS BELOW LINE AFTER TEST
+  # git_commit(git_repo, folder_path, "hello world commit", "fakeAuthor", "fakeEmail")
   error <- system2('git',
                    args = c('--git-dir',
                             git_repo,
                             '--work-tree',
                             folder_path,
-                            'add',
-                            '.'),
+                            'commit',
+                            '-m',
+                            "'hello world commit'"),
                    stdout = TRUE,
                    stderr = FALSE)
 
-  # git --git-dir sample/.git --work-tree sample commit -m 'hello world commit'
+
+  return(git_repo)
+}
+
+#' This function initializes a new Git repository in the specified folder.
+#'
+#' @param folder_path The path to the folder where the Git repository should be initialized.
+#' @return The path to the newly created Git repository.
+#' @export
+git_init <- function(folder_path) {
+  error <- system2('git',
+                   args = c('init', folder_path),
+                   stdout = TRUE,
+                   stderr = FALSE)
+  git_repo <- file.path(folder_path, '.git')
+  return(git_repo)
+}
+
+#' Git commit
+#'
+#' Git commits a file
+#'
+#' @param git_repo The git repo path
+#' @param folder_path The worktree path
+#' @param filepath The filepath we want to add
+#' @return The path to the sample .git file.
+#' @export
+git_commit <- function(git_repo, folder_path, commit_msg, author, email) {
```

Comment:
Seems the unit tests are now passing, but Check is failing. This seems to be the cause per the log:

<img width="565" alt="Screen Shot 2023-10-25 at 10 10 53 PM" src="https://github.com/sailuh/kaiaulu/assets/17270563/5e2df77a-afe6-48d3-9b64-b2e2678be5c9">